packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -183,11 +192,22 @@ export class CdkToolkit {
         let changeSet = undefined;
 
         if (options.changeSet) {
-          // only perform this check if we're going to make a changeset. This check requires permissions that --no-changeset users might not have.
-          const stackExists = await this.props.deployments.stackExists({
-            stack: stack,
-            deployName: stack.stackName,
-          });
+
+          let stackExists = false;
+          try {
+            // only perform this check if we're going to make a changeset. This check requires permissions that --no-changeset users might not have.
+            // the call should now use the lookup role, but keep behind the flag since we only need it if the changeset is being made
```

Comment:
1. We know this is kept behind the flag from reading it, you don't need this comment here.
2. "the call should now use the lookup role" - sure, but it should have only ever used that in the first place. This isn't particularly valuable here.